src/main/java/cc/reconnected/kromer/Kromer.java

```diff
@@ -160,65 +162,83 @@ public void onInitializeServer() {
+                    config.HourlyWelfare() *
+                    (config.SupporterMultiplier() * playersWithSupporter.size());
         }
+        return welfare;
+    }
 
-        float finalWelfare = Math.round(welfare * 100f) / 100f;
+    public static void executeWelfare() {
+
+        float finalWelfare = calculateWelfare();
 
         client.server
             .getPlayerList()
             .getPlayers()
             .forEach(p -> {
-                if(config.BotPermission() != null) {
-                    if(Permissions.check(p, config.BotPermission())) return; // If bot then bye
-                }
-
-                Wallet wallet = Kromer.database.getWallet(p.getUUID());
-                if (wallet == null) return;
-
-                if (
-                    Solstice.modules.getModule(AfkModule.class).get().isPlayerAfk(p)
-                ) return;
-                WelfareData welfareData = Solstice.playerData
-                        .get(p.getUUID())
-                        .getData(WelfareData.class);
-                if (
-                    !(welfareData.welfareMuted || welfareData.optedOut)
-                ) {
-                    p.sendSystemMessage(
-                        Locale.use(Locale.Messages.WELFARE_GIVEN, finalWelfare)
-                    );
-                }
-                if(!welfareData.optedOut) {
-                    CompletableFuture
-                            .supplyAsync(() -> GiveMoney.execute(config.KromerKey(), finalWelfare, wallet.address), NETWORK_EXECUTOR)
-                            .thenCompose(f -> f);
-                }
+               executeWelfareForPlayer(p,finalWelfare);
             });
     }
+    private static void executeWelfareForPlayer(ServerPlayer player, float baseWelfare) {
+        if (baseWelfare == 0) {
+            return;
+        }
+        if(config.BotPermission() != null) {
+            if(Permissions.check(player, config.BotPermission())) return; // If bot then bye
+        }
+        Wallet wallet = Kromer.database.getWallet(player.getUUID());
+        if (wallet == null) return;
+
 
+        WelfareData welfareData = Solstice.playerData
+                .get(player.getUUID())
+                .getData(WelfareData.class);
+        int interval = 3600; // one hour, in seconds, since rcc-kromer does hourly welfare.
+        AfkPlayerData solsticeData = Solstice.modules
+                .getModule(AfkModule.class)
+                .get()
+                .getPlayerData(player.getUUID());
+        int activeTime = solsticeData.activeTime;
+        var oldActiveTime = welfareData.oldActiveTime;
+        float deltaActiveTime = activeTime - oldActiveTime;
+        float relativeDeltaActiveTime = deltaActiveTime / interval; // make it 1 if equal to interval, or greater if they log off and rejoin mid-interval
+        float finalWelfare = Math.round((baseWelfare * relativeDeltaActiveTime)*100f)/100f;
```

Comment:
5 lines to calculate a single value. Why? Just make it a oneliner, as I'm pretty sure all of these values are pretty easy to understand without needing explicit names for each of them.

Reply:
Yeah, just merge this in. I don't really care that much about code style to not merge this, as you can probably see from rest of the project.